Mathlib/NumberTheory/LSeries/DirichletContinuation.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2024 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import Mathlib.NumberTheory.LSeries.ZMod
+import Mathlib.NumberTheory.DirichletCharacter.Basic
+
+/-!
+# Analytic continuation of Dirichlet L-functions
+
+We show that if `χ` is a Dirichlet character `ZMod N → ℂ`, for a positive integer `N`, then the
+L-series of `χ` has meromorphic continuation (away from a pole at `s = 1` if `χ` is trivial).
+
+All definitions and theorems are in the `DirichletCharacter` namespace.
+
+## Main definitions
+
+* `LFunction χ s`: the L-function, defined as a linear combination of Hurwitz zeta functions.
+
+## Main theorems
+
+* `LFunction_eq_LSeries`: if `1 < re s` then the `LFunction` coincides with the naive `LSeries`.
+* `differentiable_LFunction`: if `χ` is nontrivial then `LFunction χ s` is differentiable
+  everywhere.
+-/
+
+open Complex
+
+namespace DirichletCharacter
+
+variable {N : ℕ} [NeZero N]
+
+/--
+The unique meromorphic function `ℂ → ℂ` which agrees with `∑' n : ℕ, χ n / n ^ s` wherever the
+latter is convergent. This is constructed as a linear combination of Hurwitz zeta functions.
+
+Note that this is not the same as `LSeries χ`: they agree in the convergence range, but
+`LSeries χ s` is defined to be `0` if `re s ≤ 1`.
+ -/
+noncomputable def LFunction (χ : DirichletCharacter ℂ N) (s : ℂ) : ℂ := ZMod.LFunction χ s
+
+/-- The L-function of the (unique) Dirichlet character mod 1 is the Riemann zeta function. -/
+lemma LFunction_mod_one {χ : DirichletCharacter ℂ 1} :
```

Comment:
Should we not be using `lfunction` in lemma names by the naming convention? (Or namespacing)

Reply:
This is a counterpart of the existing `DirichletCharacter.LSeries_modOne_eq`.